src/Choices/Catalog.php

```diff
@@ -10,12 +10,45 @@
 
 namespace BitBag\SyliusCatalogPlugin\Choices;
 
-final class Catalog
+use Symfony\Component\Finder\Exception\DirectoryNotFoundException;
+use Symfony\Component\Finder\Finder;
+
+final class Catalog implements CatalogInterface
 {
-    public static function getTemplates(): array
+    private string $projectDir;
+
+    private string $templatesDir;
+
+    private const DEFAULT_TEMPLATE = ['default' => '@BitBagSyliusCatalogPlugin/Catalog/Templates/showProducts.html.twig'];
+
+    public function __construct(string $projectDir, string $templatesDir)
     {
-        return [
-            'bitbag_sylius_catalog_plugin.ui.form.catalog.default_template' => '@BitBagSyliusCatalogPlugin/Catalog/Templates/showProducts.html.twig',
-        ];
+        $this->projectDir = $projectDir;
+        $this->templatesDir = $templatesDir;
+    }
+
+    public function getTemplates(): array
+    {
+        $finder = new Finder();
+        try {
+            $finder->files()->in($this->projectDir.'/templates/'.$this->templatesDir)->name('*.html.twig')->depth(0);
```

Comment:
It breaks Demeter's law

Reply:
[Finder](https://github.com/symfony/finder/blob/5.4/Finder.php) methods return `$this`, so I think I can chain methods like in doctrine without braking Demeter's law. Should I make private methods for getting files?

Reply:
@senghe it's builder